transports/websocket-websys/src/lib.rs

```diff
@@ -35,7 +37,17 @@ use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Mutex;
 use std::{pin::Pin, task::Context, task::Poll};
 use wasm_bindgen::{prelude::*, JsCast};
-use web_sys::{window, CloseEvent, Event, MessageEvent, WebSocket};
+use web_sys::{CloseEvent, Event, MessageEvent, WebSocket};
+
+use crate::web_context::WebContext;
+
+/// Arbitrary, maximum amount we are willing to buffer before we throttle our user.
+const MAX_BUFFER: usize = 1024 * 1024;
+
+thread_local! {
+    /// Global web context for abstracting the `Window` vs the `WorkerGlobalScope` API
+    static GLOBAL_WEB_CONTEXT: WebContext = WebContext::new();
+}
```

Comment:
Why do we need the thread-local? We only call this in the constructor and `Drop` impl so I don't think we'd be able to measure a performance benefit here.